tests/test_generators.py

```diff
@@ -437,3 +437,33 @@ def test_correct_ratio_fm(self):
         total = len(bqm.quadratic)
         fm = sum([val == -1 for val in bqm.quadratic.values()])
         self.assertAlmostEqual(fm / total, 0.3, places=2)
+
+
+class TestKnapsack(unittest.TestCase):
+
+    def test_model(self):
+        num_items = 10
+        cqm = dimod.generators.knapsack(num_items=num_items)
+        self.assertEqual(len(cqm.variables), num_items)
+        self.assertEqual(len(cqm.constraints), 1)
+
```

Comment:
Can we also test that (some of) the solutions have the correct energy and (some) infeasible solutions do not? You can test the energy per-component.

`QM.energy` was added as part of https://github.com/dwavesystems/dimod/pull/880